### PR TITLE
[1/x] Prepare code sprint and implement repository type detection

### PR DESCRIPTION
**Pull request by felixsch**
## Description

This PR is part of our code sprint: https://trello.com/c/hU27QHob/3134-rdeb-sprint-tracker

**The whole changeset is:**
[[1/x] Prepare code sprint and implement repository type detection](https://github.com/SUSE/rmt/pull/1056)
[[2/x] Add debian mirroring support to RMT](https://github.com/SUSE/rmt/pull/1057)

**Implementations in this PR:**

- Move current `mirror.rb` into `mirror/repomd.rb` and make CLI and everything working with the new layout
- Implement `detect_repository_type` (see: https://github.com/SUSE/rmt/pull/1056/files#diff-ab471d2d1e1090f261618b78f4d678e711afb8063a7303187375a3bbe3bee075R15)

The repository type is not yet used in any way and is only there for future use.

**How to test this pull request:**

*Make sure mirroring is working as expected*
```bash
$ bin/rmt-cli mirror
# Expect: mirroring to work normally
```

*Test repository type detection*
```ruby
# Sync any of our test accounts or our RMT reference account
$ rails c
> debian = Repository.find_by(name: 'Debian-10-SUSE-Manager-Tools')
> repomd = Repository.find_by(name: 'SLE-Product-SLED15-SP6-Pool')
> debian_mirror = RMT::Mirror.new(repository: debian, logger: nil, base_dir: '/tmp')
> repomd_mirror = RMT::Mirror.new(repository: repomd, logger: nil, base_dir: '/tmp')


> repomd_mirror.detect_repository_type
# expect: to return :repomd

> debian_mirror.detect_repository_type
# expect: to return :debian
```

**Thanks for reviewing this pull request!**

If you have any question please do not hesitate to approach me! :rocket:

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/348940/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged d3c93d1e6d06c8b68037defe46e83e6452a8a1af into 803d2b255cb03f78b6f794e567510d87df12a9e5)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/348941/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged a4ed30149dd7f285b1f0afa56c2b17c0955f3770 into 803d2b255cb03f78b6f794e567510d87df12a9e5)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/349082/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 659d0e0501f5e249dd7a989e7c6cd71295ce1c4d into 803d2b255cb03f78b6f794e567510d87df12a9e5)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/350246/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 5e777366ff15b861a329a33a29849e9caeb54926 into 98f1142f2bdc0a220c50f0d1c4dacfd51e30db49)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/350456/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6d3afc2849100342870bb869df8cf7c498b7f54a into 98f1142f2bdc0a220c50f0d1c4dacfd51e30db49)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/351236/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 530de83ee0a207d355272f457811cc6384adaa5c into 7e68170344b9747f53da348fea62e2668c3f2aee)

File: lib/rmt/cli/mirror.rb [hunk 1]
```diff
@@ -79,7 +79,13 @@ def product(*targets)
   protected
 
   def mirror
-    @mirror ||= RMT::Mirror.new(logger: logger, mirror_src: RMT::Config.mirror_src_files?)
+    config = {
+      logger: logger,
+      mirroring_base_dir: RMT::DEFAULT_MIRROR_DIR,
+      mirror_src: RMT::Config.mirror_src_files?,
+      airgap_mode: false
```
Comment (digitaltom) [inline lib/rmt/cli/mirror.rb:86]: Was it a bug that we didn't set `airgap_mode` before?

File: lib/rmt/mirror.rb [hunk 2]
```diff
@@ -1,205 +1,38 @@
...
-    # we need this extra step to ensure that we write to the public directory
-    base_dir = mirroring_base_dir
-    base_dir = File.expand_path(File.join(mirroring_base_dir, '/../')) if mirroring_base_dir == RMT::DEFAULT_MIRROR_DIR
-
-    repository_dir = File.join(base_dir, '/suma/')
-    mirroring_paths = {
-      base_url: URI.join(repository_url),
-      base_dir: repository_dir,
-      cache_dir: repository_dir
+  def detect_repository_type
```
Comment (digitaltom) [inline lib/rmt/mirror.rb:15]: Imho the 'detect_' prefix is superfluous.

**Review by digitaltom — APPROVED** [review]
Great :+1:

File: lib/rmt/cli/mirror.rb (hunk 1 above)
Comment (felixsch) [reply lib/rmt/cli/mirror.rb:86]: It defaults to to false. For the sake of being explicit  we added it here

File: lib/rmt/mirror.rb (hunk 2 above)
Comment (felixsch) [reply lib/rmt/mirror.rb:15]: I agree, we keep it in mind and change the naming when we switch to our new implementations!

**Pull request merged**